controller/linodemachine_controller.go

```diff
@@ -352,6 +352,18 @@ func (r *LinodeMachineReconciler) reconcileInstanceCreate(
 		conditions.MarkTrue(machineScope.LinodeMachine, ConditionPreflightConfigured)
 	}
 
+	if machineScope.LinodeMachine.Spec.Configuration != nil && machineScope.LinodeMachine.Spec.Configuration.Kernel != "" {
```

Comment:
given I think it will cause a much larger refactor to reduce complexity here I would be ok with adding a nolint to this check for this PR